actions/ctl.go

```diff
@@ -113,36 +120,54 @@ func (a *ctlFn) Evaluate(_ rules.RuleMetadata, txS rules.TransactionState) {
 	case ctlForceRequestBodyVariable:
 		val, ok := parseOnOff(a.value)
 		if !ok {
-			tx.WAF.Logger.Error("[ctl:ForceRequestBodyVariable] unknown value %q", a.value)
+			tx.WAF.Logger.Error().
+				Str("ctl", "ForceRequestBodyVariable").
+				Str("value", a.value).
+				Msg("Unknown value")
 			return
 		}
 		tx.ForceRequestBodyVariable = val
-		tx.WAF.Logger.Debug("[ctl:ForceRequestBodyVariable] Forcing request body var with CTL to %s", val)
+		tx.WAF.Logger.Debug().
+			Str("ctl", "ForceRequestBodyVariable").
+			Bool("value", val).
+			Msg("Forcing request body var")
 	case ctlRequestBodyAccess:
 		val, ok := parseOnOff(a.value)
 		if !ok {
-			tx.WAF.Logger.Error("[ctl:RequestBodyAccess] unknown value %q", a.value)
+			tx.WAF.Logger.Error().
+				Str("ctl", "RequestBodyAccess").
+				Str("value", a.value).
+				Msg("Unknown value")
 			return
 		}
 		tx.RequestBodyAccess = val
 	case ctlRequestBodyLimit:
 		limit, err := strconv.ParseInt(a.value, 10, 64)
 		if err != nil {
-			tx.WAF.Logger.Error("[ctl:RequestBodyLimit] Incorrect integer CTL value %q", a.value)
+			tx.WAF.Logger.Error().
```

Comment:
This is missing `Msg` so won't be outputed - it seems that zerolog has `Send()` for msg-less logs.

Note that this is the trickiness of the zerolog API that achieves minimum allocations with the chance of not actually logging by mistake - it's definitely much harder to use than something like zap. But I guess we can deal with it. If there's not a linter for it yet I guess we could write it

Reply:
Ah nice, there's an official one

https://github.com/rs/zerolog/blob/master/cmd/lint/README.md

Reply:
Lint sounds good. BTW, one of the things I have been doing lately is to cover logs with unit tests but linter seems easy. Let me open an issue for that.